pkg/cmd/variable/list/list.go

```diff
@@ -114,7 +127,7 @@ func listRun(opts *ListOptions) error {
 		return fmt.Errorf("failed to get variables: %w", err)
 	}
 
-	if len(variables) == 0 {
+	if len(variables) == 0 && opts.Exporter == nil {
```

Comment:
Could explain why this is needed?

https://github.com/rajhawaldar/cli/blob/976b84afd2ce02bb575df1269baac33a0618ed9b/pkg/cmd/secret/list/list.go#L147-L149

Reply:
Added to make the return value an empty array.  example `pr list` `issue list`

https://github.com/cli/cli/blob/586eb8b865ea539832716a3d8c5cb42b11ee784c/pkg/cmd/pr/list/list.go#L179-L181

```
$ gh pr list --json id
[]
```

However, secret list does not return an empty array. 

`secret list` and `variable list` have similar usage scenarios, it would be better to align their behavior with that of secret list.

```
$ gh secret list --json name
no secrets found
```